main.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+  # my_each()
+  def my_each()
+    each do |item|
+      yield item if block_given?
+    end
+  end
+  # ending my_each()
+
+  # my_each_with_index()
+  def my_each_with_index()
+    index = 0
+    each do |item|
+      yield(item, index) if block_given?
+      index += 1
+    end
+  end
+  # my_each_with_index()
+
+  # my_select
+  def my_select()
+    new_arr = []
+    each do |item|
+      new_arr << item if yield(item)
+    end
+    new_arr
+  end
+  # my_select
+
+  # my_all?
+  def my_all?()
+    flag = true
+    each do |item|
+      feedback = yield(item)
+      flag = false if (feedback == false) || feedback.nil?
+    end
+    flag
+  end
+  # my_all?
+
+  # my_any
+  def my_any?()
+    flag = false
+    each do |item|
+      flag = true if yield(item)
+    end
+    flag
+  end
+  # my_any
+
+  # my_none?
+  def my_none?()
+    is_none = true
+    each do |item|
+      is_none = false if yield(item)
+    end
+    is_none
+  end
+  # my_none?
+
+  # my_count
+  def my_count
+    index = 0
+    each do |_item|
+      index += 1
+    end
+    index
+  end
+  # my_count
```

Comment:
- [ ] my_count counts the number of items in enum that are equal to item if an argument is given

- [ ] my_count counts the number of elements yielding a true value if a block is given